app/src/main/java/de/westnordost/streetcomplete/data/QuestType.kt

```diff
@@ -7,8 +7,11 @@ interface QuestType<T> {
     /** the icon resource id used to display this quest type on the map */
     val icon: Int
 
-    /** the title resource id used to display the quest's question */
-    val title: Int
+    /** returns title resource for when the element has the specified [tags]. The tags are unmodifiable */
+    fun getTitleBaseString(tags: Map<String, String>): Int
```

Comment:
Requesting these changes:

1. Please leave `val title: Int` in QuestType as is. Only quests based on OSM elements (`OsmElementQuestType`) should have a `getTitle(Map<String, String>)` because only OSM elements have tags. For example, quests based on notes do not have tags.
2. Please leave the name `getTitle(Map<String, String>)` as is. It is shorter, and descriptive enough.

Not entirely necessary, just suggestions:
1. `getTitleReplacements` could be renamed to `getTitleArgs`. It is shorter and the meaning is still carried over, what do you think? Naming is consistent with for example [here](https://docs.oracle.com/javase/7/docs/api/java/util/Formatter.html#format(java.util.Locale,%20java.lang.String,%20java.lang.Object...)), [here ](https://developer.android.com/reference/android/database/sqlite/SQLiteDatabase.html#query(boolean,%20java.lang.String,%20java.lang.String[],%20java.lang.String,%20java.lang.String[],%20java.lang.String,%20java.lang.String,%20java.lang.String,%20java.lang.String)) or [here](https://developer.android.com/reference/android/content/res/Resources.html#getString(int,%20java.lang.Object...))
2. in `getTitleReplacements`, the parameter name for the feature name is `typeName`. "Type" could mean different things, especially in programming context. I think `featureName` is better because the dictionary is also called `FeatureNameDictionary` (consistency) and I chose that name for that class not by accident. See [Features](https://wiki.openstreetmap.org/wiki/Features).

Reply:
> Please leave the name getTitle(Map<String, String>) as is. It is shorter, and descriptive enough.

Done.